### ISSUE TITLE
AsciiDoc code snippets in documentation

### ISSUE THREAD

**Issue by Michael1993** [junit-pioneer/junit-pioneer#335]
Demonstrated in #334, we can use a neat built-in feature of AsciiDoc to include live code examples in our `.adoc` documentation.

The proper way to utilize this feature warrants a discussion.

**Comment by Bukama** [comment]
To copy my concerns from the draft into the discussion:

> * How do you limit to (about) 60 chars per line? (see #317 / #318)
> * I'm not sure if this is helpfull. We often make comments in the documentation example, but not in the test cases

**Comment by Michael1993** [comment]
> * How do you limit to (about) 60 chars per line?

We could use Java::Geci to format code snippets before they are inserted into the documentation.

> * I'm not sure if this is helpful. We often make comments in the documentation example but not in the test cases.

Yes, this is certainly a drawback. Some ideas about what we could do to counteract this problem:
 * We could put the comments in the actual tests so they get copied with the snippet
 * We could put the content of the comments around the snippet (e.g.: instead of saying "here we call this because", we write after the test "notice how there we called X. We did this because...")

We might decide that the fruit ain't worth the squeeze, but we won't know for sure unless we try squeezing first.

**Comment by Bukama** [comment]
@Michael1993 Could you summarize your yesterdays question and the answer to it please.

**Comment by Michael1993** [comment]
@nipafx wrote on Sep 29:
> I really like this approach to showing code snippets because we can always be sure that they actually work. @Bukama's concerns are valid, though. Tests and code snippets have very different concerns and we should hence separate them.
>
> I propose to create a new source set demo (as src/demo/java) and put all the demos there. They must be written as proper tests, so we can execute them during the build, but they can be written and styled according to the documentation's needs.
>
> If possible, we could even use Spotless to limit line length to 60 chars (if we can have different configurations for different folders - I think we already had that request once).

We agreed that this was a good approach and we should play around with it in a PR. I'll go ahead when I have the time.
